Lib/posixpath.py

```diff
@@ -241,43 +226,37 @@ def expanduser(path):
         tilde = b'~'
     else:
         tilde = '~'
-    if not path.startswith(tilde):
+    if path[:1] != tilde:
         return path
     sep = _get_sep(path)
     i = path.find(sep, 1)
     if i < 0:
         i = len(path)
-    if i == 1:
-        if 'HOME' not in os.environ:
-            try:
-                import pwd
-            except ImportError:
-                # pwd module unavailable, return path unchanged
-                return path
-            try:
-                userhome = pwd.getpwuid(os.getuid()).pw_dir
-            except KeyError:
-                # bpo-10496: if the current user identifier doesn't exist in the
-                # password database, return the path unchanged
-                return path
-        else:
-            userhome = os.environ['HOME']
-    else:
+    if i != 1:
         try:
             import pwd
-        except ImportError:
+            name = path[1:i]
+            if isinstance(name, bytes):
+                name = name.decode('ascii')
+
+            userhome = pwd.getpwnam(name).pw_dir
+        except (ImportError, KeyError):
             # pwd module unavailable, return path unchanged
+            # bpo-10496: if the user name from the path doesn't exist in the
+            # password database, return the path unchanged
             return path
-        name = path[1:i]
-        if isinstance(name, bytes):
-            name = str(name, 'ASCII')
+    elif 'HOME' not in os.environ:
         try:
-            pwent = pwd.getpwnam(name)
-        except KeyError:
-            # bpo-10496: if the user name from the path doesn't exist in the
+            import pwd
+            userhome = pwd.getpwuid(os.getuid()).pw_dir
+        except (ImportError, KeyError):
```

Comment:
Previously it was clear for a reader of the code exactly which statement might raise `ImportError` and exactly which statement might raise `KeyError`. By grouping them together into a single `try`/`except` block, we lose that clarity.

Does grouping them together speed things up here? How much by?